### PR TITLE
Query Multiple tables

### PR DESCRIPTION
**Pull request by Mofuhidy**
# What I did in this PR :

- Add Constraints Foreign keys;
- Add Tables;
- Query multiple tables;

------------------------------------------------------------------------

- What animals belong to Melody Pond?
Screenshot 2023-08-15 at 15 46 44

------------------------------------------------------------------------

- List of all animals that are  pokemon (their type is Pokemon)

Screenshot 2023-08-15 at 15 47 03

------------------------------------------------------------------------

- List all owners and their animals, remember to include those that don't own any animal.
Screenshot 2023-08-15 at 15 47 15

------------------------------------------------------------------------

- How many animals are there per species?

Screenshot 2023-08-15 at 15 47 27

------------------------------------------------------------------------

- List all Digimon owned by Jennifer Orwell.
Screenshot 2023-08-15 at 16 32 16

------------------------------------------------------------------------

- List all animals owned by Dean Winchester that haven't tried to escape.
Screenshot 2023-08-15 at 15 47 44

------------------------------------------------------------------------
- Who owns the most animals?
Screenshot 2023-08-15 at 15 47 51

------------------------------------------------------------------------

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -55,3 +55,44 @@ SELECT neutered, SUM(escape_attempts) AS total_attempts FROM animals GROUP BY ne
...
+-------------------------------------------------------
+-- How many animals are there per species?
+SELECT s.id, s.name as species, COUNT(a.id) as number_of_animals
+FROM animals a JOIN species s ON a.species_id = s.id
+GROUP BY s.id;
+-----------------------------------------------------------
+-- List all Digimon owned by Jennifer Orwell.
+SELECT a.id, a.name
+FROM animals a
+JOIN owners o ON a.owner_id = o.id WHERE o.full_name = 'Jennifer Orwell';
```
Comment (OyePriscilla) [inline queries.sql]: - [x] One more thing please, the specie name (`digimon`) is missing from the query, this made you to get incorrect answer. Kindly implement this and update the screenshot. 👍

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @Mofuhidy 👍 👍, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Used primary key & foreign key mechanism for joining tables. ✔️ 
Queried multiple tables. ✔️ 
Prepared complex queries that answer analytical questions. ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql [hunk 2]
```diff
@@ -55,3 +55,45 @@ SELECT neutered, SUM(escape_attempts) AS total_attempts FROM animals GROUP BY ne
...
+-- How many animals are there per species?
+SELECT s.id, s.name as species, COUNT(a.id) as number_of_animals
+FROM animals a JOIN species s ON a.species_id = s.id
+GROUP BY s.id;
+-----------------------------------------------------------
+-- List all Digimon owned by Jennifer Orwell.
+SELECT a.id, a.name, o.full_name, s.name as species FROM animals a
+JOIN owners o ON a.owner_id = o.id
+JOIN species s ON a.species_id = s.id
+WHERE o.full_name = 'Jennifer Orwell' AND s.name = 'Digimon';
```
Comment (Gambit142) [inline queries.sql:88]: - [x] Great work implementing the change, but as the previous reviewer mentioned, the output result which is evident through the screenshot is wrong. Kindly update the screenshot as proof that this query returns the correct result. 👍 😎

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @Mofuhidy 👋, 

Great work on making the changes requested by a previous reviewer 👏 

### Highlights
- Everything that creates or changes the structure of the table was saved in the _**schema.sql**_ file ✔️
- Everything that populates tables with data was saved in the _**data.sql**_ file ✔️ 
- Everything that is a query that helped to analyze data was saved in the _**queries.sql**_ file ✔️

You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer and addressing them will improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Gambit142 — APPROVED** [review]
Hi @Mofuhidy 👋,

## Status: Approved 🟢   

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

### Highlights
- Everything that creates or changes the structure of the table was saved in the _**schema.sql**_ file ✔️
- Everything that populates tables with data was saved in the _**data.sql**_ file ✔️ 
- Everything that is a query that helped to analyze data was saved in the _**queries.sql**_ file ✔️

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**